dev-packages/e2e-tests/test-applications/nestjs-fastify/tests/transactions.test.ts

```diff
@@ -49,40 +49,6 @@ test('Sends an API route transaction', async ({ baseURL }) => {
   expect(transactionEvent).toEqual(
     expect.objectContaining({
       spans: expect.arrayContaining([
-        {
-          span_id: expect.stringMatching(/[a-f0-9]{16}/),
-          trace_id: expect.stringMatching(/[a-f0-9]{32}/),
-          data: {
-            'sentry.origin': 'manual',
-            'fastify.type': 'middleware',
-            'plugin.name': 'fastify -> @fastify/middie',
```

Comment:
do we have no more plugin/middleware spans like this anymore? Is this "OK"?

Reply:
Checked again after the updates. We still have the `middie` spans, I added them back to the tests. But the problem is now that we don't have access to the `requestHook` anymore, the spans are not formatted correctly in NestJS applications (when `setupFastifyErrorHandler` is not used). We attempt using `addFastifySpanAttributes` inside `setupFastifyErrorHandler`.

Reply:
I'm now trying to use `diagnosticsChannel` for it. Maybe we can reach the context from it

Reply:
can we try to PR something like the requestHook to the otel instrumentation, maybe? Or is this something they do not want?

Reply:
@mydea, they released a way to reach the spans: https://github.com/fastify/otel/releases/tag/v0.6.0

The problem is that with this release, they removed the vendored types we added previously. Adding `fastify` here as a `devDependency` works for TypeScript 5, but not in previous versions. Should we make this an exception for TS-3.8 support/tests?

Reply:
> they removed the vendored types

what do you mean, can we not replicate this anymore? 😢

Reply:
Unfortunately, yes: https://github.com/fastify/otel/pull/46/files#diff-7aa4473ede4abd9ec099e87fec67fd57afafaf39e05d493ab4533acc38547eb8

> Had to remove the previously vendored types due to conflicts with interface augmentation typical in fastify plugins.

Reply:
@mydea - Looks like they have overridden the types of `fastify`. So, it works now without adding `fastify` as a dependency. Please bear with me when I generate sample events and post links.